src/core/libmaven/csvreports.h

```diff
@@ -22,122 +20,65 @@ class CSVReports {
     *such as rt,mz, samples used, compound it, formula, etc
     */
 public:
-    /**
-    *empty constructor
-    */
+    enum ExportType{ GroupExport, PeakExport};
     CSVReports();
+
     /**
-    *@brief-    creating CSVReports by all the samples uploaded
-    *@see -     see details in its definition
-    */
-    CSVReports(vector<mzSample*>& insamples);
+     * CSV exporting is a four step process-
+     * 1. Create one instance of this class by giving required value of
+     * all parameters in below constructor.      
+     * 2. Add all groups one by one to be exported by calling addItem method of this class
+     * 3. call exportGroup() method of this class which will return true if report is successful
+     *  otherwise false
+     * 4. If returned value from exportGroup() is false call getErrorReport method to get error message
+     */ 
     /**
-    *@brief-    destructor, just close all open output files opened for writing csv or tab file
-    */
+     * Note: don't use same instance of this class to export more that one csv file.
+     * Otherwise, it'll export mixed groups.
```

Comment:
Why will it have mixed groups?

Reply:
@rish9511 It can have groups from earlier export, because current implementation does not have method to clear groups from that variable.

Reply:
@IpankajI  my point was that we should have groups from earlier export at all.
CsVReports should be provided the groups that have to be written to CSV.
Why should CsVReports contain a vector of groups in first place ? because if we do that, then there is overhead of clearing groups every time we are done exporting.

Reply:
@rish9511 There are places in Qt-code where we need to export filtered groups which are added one at a time based on filtering criteria. Again, if we are adding group one at a time and writing that group directly to file then file stream has to be remained open for long time which can result in file write exception.